File: soma_v1.java
/*primeira versão da função soma*/
public class Main
{
	public static void main(String[] args) {
		
		int a = 5;
		int b = 5;
		
		System.out.print(soma (a,b)); //chamo a função soma ao mesmo tempo passando os valores como paramentro

	}
	
	public static int soma (int a, int b){
	    int c = 0; // crio uma variavel local para receber o valor
	    c = a + b; // realizo a operação de soma
	    return c; // retorno o resultado
	}
}

File: soma_v1.cs
/******************************************************************************
primeira versão do programa soma
*******************************************************************************/
using System;
class Operacao {
  static void Main() {
    
    //atribuição de dois valores as variaveis a e b
    int a = 5;
    int b = 5;
    
    //chamo a função e ao mesmo tempo passo as variaveis
      Console.WriteLine (soma(a,b));
  }
  
  static int soma (int a, int b){
      int resultado = a + b;
      return resultado;
  }
}